server/odc-core/src/main/java/com/oceanbase/odc/core/authority/permission/ProjectPermission.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.core.authority.permission;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.Validate;
+
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.core.authority.model.SecurityResource;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+/**
+ * @Author: Lebie
+ * @Date: 2024/11/11 19:26
+ * @Description: []
+ */
+@Getter
+public class ProjectPermission extends ResourcePermission {
```

Comment:
Because the `actions` of "Global Project Role" differ significantly from the existing `ResourcePermission` `actions`, it is necessary to reimplement the `implies` method. `ResourcePermission` `actions` have implicit relationships, whereas Global Project Role actions are specific project roles like `DBA`, `OWNER`, `DEVELOPER`, etc., with no implicit relationships between them.